src/components/Single-Article.jsx

```diff
@@ -25,12 +28,47 @@ export const SingleArticle = () => {
             setComments(body.comments)
         })
     }, [article_id])
+
+    useEffect(() => {
+        getUsers()
+        .then((body) => {
+            setUsers(body.users)
+    })
+    }, [])
     
     const handleVote = (article_id, vote) => {
             patchVotes(article_id, vote);
             setVoteChanges(prev => ({...prev, [article_id]: vote}));
             }
 
+    const handleUsername = (event) => {
+        setComment({ ...comment, username: event.target.value})
+    }
+
+    const handleBody = (event) => {
+        setComment({ ...comment, body: event.target.value})
+    }
+
+    const handleComment = (event) => {
+        event.preventDefault();
+        setSubmitting(true);
+        const newComment = {
+            username: comment.username,
+            body: comment.body
+        };
+        postComment(article_id, newComment)
+            .then((newComment) => {
+                setComments((prevComments) => [newComment, ...prevComments]);
```

Comment:
consider doing this optimistically: https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering